pkg/crd/testdata/cronjob_types.go

```diff
@@ -552,6 +553,30 @@ func (u *URL2) String() string {
 	return (*url.URL)(u).String()
 }
 
+// URL3 wraps [net/url.URL]. It implements [encoding.TextMarshaler] so that it
+// can be used in K8s CRDs such that the CRD resource will have the URL but
+// operator code can can work with the URL struct.
+type URL3 struct{ url.URL }
+
+var _ encoding.TextMarshaler = (*URL3)(nil)
+
+// MarshalText implements [encoding.TextMarshaler].
+func (u *URL3) MarshalText() (text []byte, err error) {
+	return u.MarshalBinary()
+}
+
+// URL4 is an alias of [net/url.URL]. It implements [encoding.TextMarshaler] so
+// that it can be used in K8s CRDs such that the CRD resource will have the URL
+// but operator code can can work with the URL struct.
+type URL4 url.URL
```

Comment:
I do not understand the difference between the two, could you explain why they both exist?

What seems to be missing as a testcase is something that implements both json marshaller and text marshaller, in that case the json marshalling must take precedence

Reply:
> I do not understand the difference between the two, could you explain why they both exist?

It's been a while since I wore that piece of code. Need to figure out myself 🙃

> What seems to be missing as a testcase is something that implements both json marshaller and text marshaller, in that case the json marshalling must take precedence

Will do a rebase and add the test in the next few days I hope.

Reply:
> I do not understand the difference between the two, could you explain why they both exist?

You mean `URL4` and `URL4`? I mimicked the `URL` and `URL2` types. One is a regular struct, the other is a newtype. I guess it makes sense to have both, as they might behave differently when using reflection?

> What seems to be missing as a testcase is something that implements both json marshaller and text marshaller, in that case the json marshalling must take precedence

Added.